src/pydata_sphinx_theme/__init__.py

```diff
@@ -49,33 +50,58 @@ def update_config(app, env):
-        if "G-" in gid:
-            gid_js_path = f"https://www.googletagmanager.com/gtag/js?id={gid}"
-            gid_script = f"""
-                window.dataLayer = window.dataLayer || [];
-                function gtag(){{ dataLayer.push(arguments); }}
-                gtag('js', new Date());
-                gtag('config', '{gid}');
-            """
-        # In this case it is "old-style" google analytics
-        else:
-            gid_js_path = "https://www.google-analytics.com/analytics.js"
-            gid_script = f"""
-                window.ga = window.ga || function () {{
-                    (ga.q = ga.q || []).push(arguments) }};
-                ga.l = +new Date;
-                ga('create', '{gid}', 'auto');
-                ga('set', 'anonymizeIp', true);
-                ga('send', 'pageview');
-            """
+        msg = (
+            "'google_analytics_id' is deprecated and will be removed in "
+            "version 0.11, please refer to the documentation "
+            "and use 'analytics' instead."
+        )
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+        analytics.update({"google_analytics_id": gid})
 
-        # Link the JS files
-        app.add_js_file(gid_js_path, loading_method="async")
-        app.add_js_file(None, body=gid_script)
+    if analytics:
+        # Plausible analytics
+        plausible_domain = analytics.get("plausible_analytics_domain")
+        plausible_url = analytics.get("plausible_analytics_url")
+
+        if plausible_domain and plausible_url:
+            plausible_script = f"""
+                data-domain={plausible_domain} src={plausible_url}
+            """
+            # Link the JS file
+            app.add_js_file(None, body=plausible_script, loading_method="defer")
+
+        # Two types of Google Analytics id.
+        gid = analytics.get("google_analytics_id")
+        if gid:
+            # In this case it is "new-style" google analytics
+            if "G-" in gid:
+                gid_js_path = f"https://www.googletagmanager.com/gtag/js?id={gid}"
+                gid_script = f"""
+                    window.dataLayer = window.dataLayer || [];
+                    function gtag(){{ dataLayer.push(arguments); }}
+                    gtag('js', new Date());
+                    gtag('config', '{gid}');
+                """
+            # In this case it is "old-style" google analytics
+            else:
+                gid_js_path = "https://www.google-analytics.com/analytics.js"
+                gid_script = f"""
+                    window.ga = window.ga || function () {{
+                        (ga.q = ga.q || []).push(arguments) }};
+                    ga.l = +new Date;
+                    ga('create', '{gid}', 'auto');
+                    ga('set', 'anonymizeIp', true);
+                    ga('send', 'pageview');
+                """
+
+            # Link the JS files
+            app.add_js_file(gid_js_path, loading_method="async")
```

Comment:
is there a reason we'd use `defer` for plausible, but `async` here?

Reply:
I copied this but apparently this is what they have in their doc. Maybe to have correct timing?? https://plausible.io/docs/plausible-script